### PR TITLE
docs: fix links in README and "overview" section, add tutorial

### PR DESCRIPTION
**Pull request by JuroOravec**
Bit of cleanup for the docs:

- Fixed links in the files in the `overview` directory
- Removed empty `/guides/cookbook/` path - I was thinking of use it, but eventually scrapped that for now
- Made the "dev guides" actually part of the site navigation (before they were hidden and accessible only through a link from the `/overview/development` page

Plus added a section to the docs for a tutorial to go over the core of how to write components. IMO looks very crisp 😄.
- There I broke down the original "Your first component" article into 4, so that each one really goes only over a single concept. Plus added an article on slots. So it goes like:
	1. Learn how to define a Component class and how to write HTML for it.
	2. Add JS and CSS to the component
	3. Learn how to register a component and how to use it in another template
	4. Learn how to parametrize your component through component inputs
	5. Add slots to the component

	I think there are 3 more topics that could be part of this tutorial (all of which are currently in the `/concepts/fundamenals` section):
	6. Render components from within Python
	7. Define a component as view
	8. Format HTML attributes with `{% html_attrs %}`

	But that can be added later.

So now the `/concepts` section is split into:
1. Getting started - the tutorial-style onboarding
6. Fundamentals - Documentation for common features / patterns
9. Advanced - Docs for advanced features

I can imagine the tutorial possibly living in other part of the docs, but I've just put it into `/concepts` for now as that's fine.

### CODE REVIEW

File: docs/concepts/fundamentals/components_in_templates.md
Comment (JuroOravec) [inline docs/concepts/fundamentals/components_in_templates.md:1]: Moved to `/concepts/getting_started/components_in_templates`

File: docs/concepts/fundamentals/your_first_component.md
Comment (JuroOravec) [inline docs/concepts/fundamentals/your_first_component.md:1]: Moved to `/concepts/getting_started/your_first_component` and updated

**Review by EmilStenstrom — APPROVED** [review]
This looks awesome! I had to install mkdocs-nav-weight manually to get this to work. Could it be missing from requirements-dev.txt?

I think we're ready to make these the official docs, and point README.md and PyPI docs to this site instead. Also remove the "Documentation is work in progress" at the top. Separate PR?

**Comment by JuroOravec** [comment]
> This looks awesome! I had to install mkdocs-nav-weight manually to get this to work. Could it be missing from requirements-dev.txt?

@EmilStenstrom Did you mean requirements-**dev**.txt or requirements-**docs**.txt? Currently all mkdocs dependencies are in requirements-**docs**.txt

> I think we're ready to make these the official docs, and point README.md and PyPI docs to this site instead. Also remove the "Documentation is work in progress" at the top. Separate PR?

Let's goooo!

TBH, I'm pretty sure there are still some broken links inside the documentation pages that are under the `concepts` or `guides` sections. But we'll get to them in due time.

**Comment by EmilStenstrom** [comment]
> > This looks awesome! I had to install mkdocs-nav-weight manually to get this to work. Could it be missing from requirements-dev.txt?
> 
> @EmilStenstrom Did you mean requirements-**dev**.txt or requirements-**docs**.txt? Currently all mkdocs dependencies are in requirements-**docs**.txt

You're right, I installed from the wrong place. 

> > I think we're ready to make these the official docs, and point README.md and PyPI docs to this site instead. Also remove the "Documentation is work in progress" at the top. Separate PR?
> 
> Let's goooo!
> 
> TBH, I'm pretty sure there are still some broken links inside the documentation pages that are under the `concepts` or `guides` sections. But we'll get to them in due time.

Ok, let's wait until then.

**Pull request merged**